bin/webpack-dashboard.js

```diff
@@ -22,11 +22,32 @@ const main = module.exports = opts => { // eslint-disable-line max-statements
   program.option("-m, --minimal", "Minimal mode");
   program.option("-t, --title [title]", "Terminal window title");
   program.option("-p, --port [port]", "Socket listener port");
+  program.option("-e, --example", "Run example");
```

Comment:
I feel like it could be a little odd for production users to see this option since they wouldn't be able to use it. Could we create a development binary that does something like this:

```js
#!/usr/bin/env node

"use strict";

const webpackDashboard = require("./webpack-dashboard");

webpackDashboard({ developmentMode: true });
```

and conditionally add the option based on the `developmentMode` option? Having a wrapper like that might allow us to move the `isExample` code outside the regular binary and inject from the wrapper instead.

Reply:
I think I see what you're saying and I like the idea a lot! I agree that having a separate `-e` flag that can't be used for folks grabbing `webpack-dashboard` from npm feels a little strained. It'd be nice to have `webpack-dashboard.dev.js` that just injects that flag.

One thing I'm thinking about – how would this affect the ability for the dev environment to be run cross-platform? What I like about the current setup is that it doesn't require any sort of permissions granting and is agnostic to a user's OS. I suppose we could run the above ^ directly with `node` in the `dev` script to avoid any of that. Do you have any other suggestions?

Reply:
I think that sounds like a good idea!

Reply:
Also, do you have any ideas about how to break that `isExample` code out? The reason we have it in there is b/c the `webpack-dashboard` CLI command spawns a child process that is responsible for running your startup script. And in this case, we basically short circuit that and say, "Hey run the example project instead." I'll experiment, but not sure how easy it'll be to pull that out without duplicating all of the `socket-io` server code.

Reply:
I was thinking maybe it could accept a `command` and `args` option,

that way you could modify the `if (logFromChild)` block to something like this:

```js
if (logFromChild || isDevelopmentMode) {
    const command = opts.command || program.args[0];
    const args = opts.args || program.args.slice(1);
    const env = process.env;

    env.FORCE_COLOR = true;

    child = spawn(command, args, {
      env,
      stdio: [null, null, null, null],
      detached: true
    });
  }
```

Reply:
You know what I realized is easier. Just changing `package.json` scripts to:

```
"dev": "cross-env EXAMPLE=duplicates-esm node bin/webpack-dashboard.js -- webpack-cli --config examples/config/webpack.config.js --watch"
```

So just actually use the way the dashboard works (supply it the command we want to run), rather than wrapping this one use case an additional CLI flag. I'll push this up now and let me know what you think.

Reply:
Yeah that's great 😁